### PR TITLE
Fix potential github action smells

### PR DESCRIPTION
**Pull request by ceddy4395**
Hey! :slightly_smiling_face: 
I want to contribute the following changes to your workflow: 

- Avoid executing  scheduled workflows on forks
- Stop running workflows when there is a newer commit in PR
- Stop running workflows when there is a newer commit in branch

(These changes are part of a research Study at TU Delft looking at GitHub Action Smells. [Find out more](https://ceddy4395.github.io/research/gha-smells.html))

### CODE REVIEW

**Comment by ilike2burnthing** [comment]
We'll also have to see how this works with:

https://github.com/Jackett/Jackett/blob/f812e49a456908b87998ece7daaaf7a53b7b6f6a/azure-pipelines.yml#L13-L14

Example:
Three commits get pushed in quick succession, pipeline build runs for the first, other two wait, actions begin to run for the first, but are cancelled because of the later commits.

Not sure if that will show up as a failure. Should be fine, but when is anything ever simple? 😋

**Comment by ceddy4395** [comment]
> Thanks for the contribution.
> 
> A couple of changes per our own code style and examples in GH docs:
> 
> * https://docs.github.com/en/actions/using-jobs/using-concurrency
> * https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution
> 
> Let me know if that all makes sense.

Thank you for the changes, they make sense to me! 
I'll take your feedback into account for my research and future changes.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -33,11 +33,16 @@ on:
...
+  
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 
 jobs:
   analyze:
     name: Analyze
     runs-on: windows-2022
+    if: github.repository == 'jackett/jackett'
```
[the remark this answers is not in the document]
Comment (ceddy4395) [reply .github/workflows/codeql-analysis.yml]: @mynameisbogdan I quickly looked it up and couldn't find anything on whether it is or isn't, so I'm assuming it is actually case-sensitive. I've fixed both if statements.

**Comment by garfield69** [comment]
v0.21.2485

**Comment by ilike2burnthing** [comment]
Got a chance to test `cancel-in-progress` and how it interacts with `batch` (as raised in https://github.com/Jackett/Jackett/pull/15274#issuecomment-2077667225) with https://github.com/Jackett/Jackett/commit/8a13f29c3e682b6b65dc52c02457a6760185d22a and https://github.com/Jackett/Jackett/commit/1724da98e7c1356e6073d035cdd288039f47178b.

The actions for the first commit are cancelled and then start running for the second. The build for the first commit continues, and once complete is published to GH, however the action checks for the commit are listed as 'not successful'.

So it seems fine, it's just useful for us to know that if we see that red X in a case like this, there's nothing to be concerned about.

**Pull request merged**